packages/interactive-messages/test/helpers.js

```diff
@@ -96,14 +97,28 @@ function delayed(ms, value, rejectionReason) {
     error = new Error(rejectionReason);
   }
   return new Promise(function (resolve, reject) {
-    var id = setTimeout(function () {
-      clearTimeout(id);
-      if (error) {
-        reject(error);
-      } else {
-        resolve(value);
-      }
-    }, ms);
+    if (ms > 0) {
+      var id = setTimeout(function () {
+        clearTimeout(id);
+        if (error) {
+          reject(error);
+        } else {
+          resolve(value);
+        }
+      }, ms);
+    } else if (ms === 0) {
+      // setImmediate allows the returned Promise to resolve much sooner than setTimeout.
+      // See: https://nodejs.org/en/docs/guides/event-loop-timers-and-nexttick/#setimmediate-vs-settimeout
```

Comment:
Thanks for having this link here. This is helpful for learning how the difference is and how `setImmediate` works.